### ISSUE TITLE
Override user entered values

### ISSUE THREAD

**Issue by aphillipo** [christianalfoni/formsy-react#247]
I want to be able to override a user set value in a text field and I'm really confused as to why my prop is refusing to update the field. I've looked and looked but it seems once you type into a field it's impossible to override the value in a declarative way?

If there is a clear way to do this or add it I would love to know!

**Comment by Semigradsky** [comment]
Something like this?

```js
function overrideValue(value) {
  return value + '42';
}

const MyInput = React.createClass({
  ...,
  changeValue(event) {
    const value = event.currentTarget.value;
    this.setValue(overrideValue(value));
  },
  render() {
    return <input ... onChange={this.changeValue} >
```

Please, provide more information.

**Comment by aphillipo** [comment]
That is not what I mean - here are the steps I expect to be able to happen.

1) I load a form with some customer data from a Redux store/action.
2) The form data is edited in various places
3) One of the actions/edits loads other parts of the form and this causes data in the store to change, which causes data in the props to changes which updates data in the form.
4) Under certain conditions I wish for the edited data to be overridden by these updated props.
5) Obviously edits take precedence over prop changes, I guess this is correct...?

Maybe in componentWillReceiveProps I can setValue on the ref of the relevant input.

**Comment by zimok** [comment]
I spent some time hacking on formsy-react (kudos to creators).
Had the same case as @aphillipo and solved in this way: 

1) load form data from store and attach it to props --> attach to form
2) attach an `onChange` event listener to every form's input change
3) form get edited by user changed
4) listeners set on `onChange` propagate changed properties to the store (you can also read current form state with `this.refs.form.getCurrentValues()`), so every input stays in sync between store and form

let me know if it can help you

**Comment by aphillipo** [comment]
Yep, this is exactly how I ended up doing things.  I have say it feels like a bit of a mess.  As I'm writing this I'm wondering if there is a form on change event that I should use instead...

**Comment by zimok** [comment]
there's a "global" `onChange` event that you can attach to the `<Formsy.Form>` component itself. Even though I had some issues using it as I needed more granular control of what changed on the form.
